components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -103,6 +103,22 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
         }.getAndConsumeRustString()
     }
 
+    /**
+     * Constructs a URL used to begin the `force_auth` OAuth flow for the requested email, scopes and keys.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @param wantsKeys Fetch keys for end-to-end encryption of data from Mozilla-hosted services
+     * @return String that resolves to the flow URL when complete
+     */
+    fun beginForceAuthOAuthFlow(email: String, scopes: Array<String>, wantsKeys: Boolean): String {
```

Comment:
Why not re-use `beginOAuthFlow` with an optional email param?

Reply:
good plan! In Kt and Swift only?

Reply:
Besides the URL params how different is the rust code?

Reply:
1 param is different (action), 1 param is new (email)

Reply:
FWIW, I can imagine a world in which the caller might want to provide an email address as a hint (say, one they captured in an about:welcome style promotion during first-run) but don't care the enforce that the user *must* authenticate as that email address.